src/pat/autotoc/autotoc.js

```diff
@@ -78,16 +78,21 @@ export default Base.extend({
         var activeId = null;
 
         $(self.options.levels, self.$el).each(function (i) {
-            const section = this.closest(self.options.section);
             const $level = $(this);
-            let id = $level.prop("id") ? $level.prop("id") : $(section).prop("id");
+            const $section = $level.closest(self.options.section);
+            let id = $section.prop("id") || $level.prop("id");
 
-            if (!id || $("#" + id).length > 0) {
```

Comment:
@thet I do not understand the logic behind this. We should not auto-create an ID if we have a unique fieldset#id but for my understanding this if is always true (no ID or an element with ID) ... not really sure how this could work before.

Additionally we can generate this always if we create a vertical TOC and assign it to the level elements to get a unique URL hash.